### PR TITLE
Add new alert target for the Openshift Prometheus Federation.

### PR DESCRIPTION
**Pull request by Boomatang**
## Additional Information

A new warning TargetDown alert specifically for the openshift-monitoring-federation scrape target needs to be created so we are notified if federation is failing.

**JIRA** https://issues.redhat.com/browse/INTLY-10161

## Verification Steps
As the verifier of the PR the following process should be done:
- install RHMI as normal
- in the openshift-monitorig edit the stateful-set for prometheus-k8s.
 - reduce the number of replicas to zero.
- go to the middleware-monitoring prometheus instance https://prometheus-route-middleware-monitoring.apps.<CLUSTER URL>/alerts
- find the alert `OpenshiftPrometheusFederationTargetDown`.
  - The alert should be in a pending state
  - After 10 minutes the alert should be firing
- Check the `TargetDown` alert
  - This should **not** be alerting
- During the uninstall this alert should not be firing. 

### Installation Verification
- Ensure the author of the PR has attached a log of the installation run from his branch to the jira or pr and check that it exited as expected.
- Verify the fresh installation is correct on cluster provided by PR author 
### Upgrade Verification
- After installation verification, notify the PR author to begin an upgrade on their cluster
- Ensure the developer of the PR has attached a log of the upgrade run from his branch to the jira or pr and check that it exited as expected. 
- If possible, look at the tasks that ran and see they match the PR
- Verify the upgrade is correct on cluster provided by PR author 

## Checklist

- [x] Updated [Changelog](https://github.com/integr8ly/installation/blob/master/CHANGELOG.md)
- [ ] Tested Installation
- [ ] Tested Uninstallation
- [ ] Tested or Created follow on for Upgrade

Is an upgrade task required and are there additional steps needed to test this?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by obrienrobert** [comment]
@Boomatang using the verification steps, the alert looks to be working as expected. I've noticed the alert message might need to be formatted/modified to be easier to understand:

OpenshiftMonitoringTargetDown

**Comment by Boomatang** [comment]
@obrienrobert Thanks for that, I have updated the alert message and if the same information from your image is used it should read. `The endpoint, prometheus-k8s.openshift-monitoring.svc:9091, is down in openshift-monitoring namespace.`

File: roles/middleware_monitoring_config/templates/target_down_alerts.yml.j2 [hunk 1]
```diff
@@ -23,4 +23,12 @@ spec:
         expr: (up{job="blackbox"}==0)
         for: 10m
         labels:
-          severity: warning
\ No newline at end of file
+          severity: warning
+      - alert: OpenshiftPrometheusFederationTargetDown
+        annotations:
+            message: >-
+                The endpoint, {{ '{{' }} $labels.instance {{ '}}' }}, is down in {{ '{{' }} $labels.namespace {{ '}}' }} namespace.
```
Comment (obrienrobert) [inline roles/middleware_monitoring_config/templates/target_down_alerts.yml.j2:30]: Great, thank for making that change! One small thing. How about changing `is down in` to `is down in the`?

Comment (Boomatang) [reply roles/middleware_monitoring_config/templates/target_down_alerts.yml.j2:30]: Do you know what I was reading it back after I force push and thought the very same thing. Couldn't decide if I should push again.

**Comment by obrienrobert** [comment]
I've had a quick look at the Prometheus docs around scrape targets:
```
In Prometheus terms, an endpoint you can scrape is called an instance, usually corresponding to a 
single process. A collection of instances with the same purpose, a process replicated for scalability 
or reliability for example, is called a job.
```
It looks like once endpoint is an instance. Maybe it would make sense to reflect this in the message, something like:

instanceDown

WDYT?

**Comment by Boomatang** [comment]
@obrienrobert It does read better as the instance and I have changed it around. I do feel also we maybe getting into the weeds about now and if there is more changes maybe it would be better to have a call about it.

**Comment by obrienrobert** [comment]
Verified this using the steps provided.

/lgtm

**Review by obrienrobert — APPROVED** [review]

**Pull request closed without merging**